app/services/data_stage/process_school_changes.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module DataStage
+  class ProcessSchoolChanges < ::BaseService
+    include GiasTypes
+
+    UNHANDLED_ATTRIBUTES = %w[
+      administrative_district_code
+      administrative_district_name
+      school_type_code
+      school_type_name
+      section_41_approved
+      ukprn
+    ].freeze
+
+    def call
+      DataStage::SchoolChange.unhandled.status_changed.find_each do |change|
+        next if has_unhandled_attributes?(change.attribute_changes)
+
+        if change.attribute_changes.key? "school_status_code"
+          handle_status_code_change(change)
+        end
+      end
+    end
+
+  private
+
+    def has_unhandled_attributes?(attributes)
+      (UNHANDLED_ATTRIBUTES & attributes.keys).any?
+    end
+
+    def handle_status_code_change(change)
+      from_code = change.attribute_changes["school_status_code"]&.first
+      to_code = change.attribute_changes["school_status_code"]&.last
+
+      if open_status_code?(from_code) && !open_status_code?(to_code)
+        close_school(change)
+      elsif !open_status_code?(from_code) && open_status_code?(to_code)
+        open_school(change)
+      else
+        # just sync then
+        change.school.create_or_sync_counterpart!
+      end
+    end
+
+    def open_school(change)
+      school = change.school
+
+      # is there a predecessor
+      last_link = school.school_links.order(created_at: :asc).last
+
+      if last_link.nil? || last_link.link_type == "Predecessor"
+        DataStage::School.transaction do
+          # no predecessor or simple predecessor - add new school
+          school.create_or_sync_counterpart!
+
+          # transfer assets if predecessor
+          if last_link.present?
+            predecessor = ::School.find_by(urn: last_link.link_urn)
+            transfer_assets_from_predecessor(school: school.counterpart, predecessor: predecessor) if predecessor.present?
+          end
+
+          change.update!(handled: true)
+        end
+      end
+    end
+
+    def close_school(change)
+      # let any related school open transfer any assets
+      # just sync attribute changes
+      DataStage::School.transaction do
+        # only sync changes don't create
+        change.school.create_or_sync_counterpart! if change.school.counterpart.present?
+        change.update!(handled: true)
+      end
+    end
+
+    def transfer_assets_from_predecessor(school:, predecessor:)
+      # move everything to the new school
+      Partnership.active.where(school: predecessor).each { |partnership| partnership.update!(school: school) }
+      # TODO: What should we do here if the school already has a school_cohort (if anything)?
```

Comment:
That sounds like a pain, because what if the school cohorts have different induction choices on them? 😢 

I think if the school cohorts are for a different cohort we should be fine to just move it.

If the cohort is the same, and if existing school cohort is 'almost identical' to predecessor school cohort, we can probably 'merge' them - just move everything pointing at old school cohort into new school cohort and delete the old one.

But if they have the same cohort, and they are different (say, different induction choice) then we are in trouble.

This actually is also a problem with partnerhsips - if a school has an active partnership, and its predecessor has an actie partnership with a different provider or delivery partner, I don't know if we should just transfer the partnership to new school :/

Reply:
Yes, this is the stuff that we need to understand. The `ProcessSchoolChanges` service isn't actually called from anywhere yet, it's a placeholder for discussion.

Reply:
At the moment I'm only trying to handle the case where a school closes and reopens, (where there is a `predecessor` but not one from a merger or split).  In this case there shouldn't be any choices made on the school as we've just created it - but there may be if it's been added/updated already manually.

Reply:
Moving this class to a separate PR